### PR TITLE
Remove usage of CWT in syntactic classifier

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
We already use this pattern in the rest of our async taggers.  But syntactic tagger has always stood alone, so it needed to be specialized here.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Classification/SyntacticTaggerTests.cs [hunk 1]
```diff
@@ -51,8 +51,6 @@ class Program2
                     workspace.GetService<IGlobalOptionService>(),
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Classification/SyntacticTaggerTests.cs:51]: view with whitspace off.

File: src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs [hunk 2]
```diff
@@ -4,494 +4,502 @@
...
+            _subjectBuffer.ChangedOnBackground += this.OnSubjectBufferChanged;
+        }
 
-            internal void IncrementReferenceCount()
+        public static TagComputer GetOrCreate(
+            SyntacticClassificationTaggerProvider taggerProvider, ITextBuffer2 subjectBuffer)
+        {
+            taggerProvider._threadingContext.ThrowIfNotOnUIThread();
+
+            if (!subjectBuffer.Properties.TryGetProperty(s_uniqueKey, out TagComputer tagComputer))
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs:109]: >TryGetProperty [](http://example.com/codeflow?start=42&length=14)

why not use GetOrCreateSingletonProperty?

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs:109]: Will do!

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.cs [hunk 3]
```diff
@@ -17,52 +17,39 @@
 using Microsoft.VisualStudio.Text.Tagging;
 using Microsoft.VisualStudio.Utilities;
 
-namespace Microsoft.CodeAnalysis.Classification
+namespace Microsoft.CodeAnalysis.Classification;
```
Comment (sharwell) [inline src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.cs:20]: ouch. not expecting this to all be in the same commit (git blame will not be helpful for figuring out the change)

**Review by genlu — APPROVED** [review]
:shipit:

**Pull request merged**